### PR TITLE
docs: add examples to the RTD

### LINKED ISSUE #1716: Add the examples/client_example to RTD
**Description of issue or feature request**:
An example can be a good start point for those reading the RTD Updater.
We could add the [client_example](https://github.com/theupdateframework/python-tuf/tree/develop/examples/client_example) in a way it can be explored by who wants to implement a client.

**Current behavior**:
The `ngclient.Updater` RTD does not have an example or point to the examples folder. The event folder is available only in the source code.

**Expected behavior**:
Easy landing or example demonstration to who wants to implement a client.

### PR DESCRIPTION
**Pull request by kairoaraujo**
This commit adds to the RTD the links references to source code
examples.
The examples are added to TUF ngclient Updater and Metadata.

Fixes #1716

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Review by lukpueh — COMMENTED** [review]
Thanks for the patch! I think it would drastically improve discoverability of the examples, if they were referenced in the [main menu](https://github.com/theupdateframework/python-tuf/blame/f172972f721aa7b2fd844302e0f3046bca2614f3/docs/index.rst#L17-L19) too, e.g.:
- API Reference
- Installation
- Usage examples
- Instructions for Contributors

We can decide later if we create a dedicated examples landing/overview page on RTD, or in a README.md in the examples directory. For the time being the menu item could simply link to the examples directory in the repo. What do you think? Would you mind updating your PR to do this?

**Comment by kairoaraujo** [comment]
> We can decide later if we create a dedicated examples landing/overview page on RTD, or in a README.md in the examples directory. For the time being the menu item could simply link to the examples directory in the repo. What do you think? Would you mind updating your PR to do this?

I think it is good. My only concern was what you already mentioned, going out from the RTD without a landing page in the examples and being out of the RTD.
I don't mind updating the PR :) 

Maybe as we don't have a README:

- API Reference
- Installation
- Usage examples
  - client code example
  - repository code example
- Instructions for Contributors

I am also ok with your suggestion.

**Comment by lukpueh** [comment]
I slightly lean towards not having sub-items in the menu in order to not convolute it. If you don't mind the effort, I'd prefer if you just added a barebones README.md to the examples folder with a similar paragraph that you already wrote for this PR.

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Usage examples
+
+* [client](client_example)
+* [repository](repo_example)
```
Comment (lukpueh) [inline examples/README.md:4]: Works for me to start with! 👍

**Review by lukpueh — APPROVED** [review]
Thanks for addressing my comments!

**Pull request merged**